sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -0,0 +1,692 @@
+              DATETIME(shutdown.submission_timestamp)
+            )
+          ),
+          (STRUCT(main.attribution_experiment, 'main', DATETIME(main.submission_timestamp)))
+        ]
+      ).earliest_value_source AS attribution_experiment__source_ping,
+      mozfun.norm.get_earliest_value(
+        [
+          (
+            STRUCT(
+              new_profile.attribution_medium,
+              'new_profile',
+              DATETIME(new_profile.submission_timestamp)
+            )
+          ),
+          (
+            STRUCT(shutdown.attribution_medium, 'shutdown', DATETIME(shutdown.submission_timestamp))
+          ),
+          (STRUCT(main.attribution_medium, 'main', DATETIME(main.submission_timestamp)))
+        ]
+      ).earliest_value_source AS attribution_medium__source_ping,
+      mozfun.norm.get_earliest_value(
+        [
+          (
+            STRUCT(
+              new_profile.attribution_source,
+              'new_profile',
+              DATETIME(new_profile.submission_timestamp)
+            )
+          ),
+          (
+            STRUCT(shutdown.attribution_source, 'shutdown', DATETIME(shutdown.submission_timestamp))
+          ),
+          (STRUCT(main.attribution_source, 'main', DATETIME(main.submission_timestamp)))
+        ]
+      ).earliest_value_source AS attribution_source__source_ping,
+      CASE
+        WHEN new_profile.client_id IS NULL
+          THEN FALSE
+        ELSE TRUE
+      END AS reported_new_profile_ping,
+      CASE
+        WHEN shutdown.client_id IS NULL
+          THEN FALSE
+        ELSE TRUE
+      END AS reported_shutdown_ping,
+      CASE
+        WHEN main.client_id IS NULL
+          THEN FALSE
+        ELSE TRUE
+      END AS reported_main_ping
+    ) AS metadata
+  FROM
+    new_profile_ping AS new_profile
+  FULL OUTER JOIN
+    shutdown_ping AS shutdown
+  USING
+    (client_id)
+  FULL OUTER JOIN
+    main_ping AS main
+  USING
+    (client_id)
+  WHERE
+    client_id IS NOT NULL
+),
+_previous AS (
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.telemetry_derived.clients_first_seen_v2`
+)
+SELECT
+  client_id,
+  COALESCE(_previous.sample_id, _current.sample_id) AS sample_id,
+  COALESCE(_previous.first_seen_date, _current.first_seen_date) AS first_seen_date,
+  COALESCE(_previous.submission_timestamp, _current.submission_timestamp) AS submission_timestamp,
+  COALESCE(_previous.architecture, _current.architecture) AS architecture,
+  COALESCE(_previous.app_build_id, _current.app_build_id) AS app_build_id,
+  COALESCE(_previous.app_display_version, _current.app_display_version) AS app_display_version,
+  COALESCE(_previous.app_name, _current.app_name) AS app_name,
+  COALESCE(_previous.vendor, _current.vendor) AS vendor,
+  COALESCE(_previous.app_version, _current.app_version) AS app_version,
+  COALESCE(_previous.xpcom_abi, _current.xpcom_abi) AS xpcom_abi,
+  COALESCE(_previous.experiments_key, _current.experiments_key) AS experiments_key,
```

Comment:
Should `_previous` experiment fields be updated with `_current`?  Do NULL experiment fields mean that the new profile was not part of an experiment?

Reply:
That's probably the case, that the new profile was not part of an experiment.
Then the question would be, is it the requirement to retrieve the first experiment that each client was part of, or only if it had an experiment on or near the first_seen_date? cc/ @irrationalagent

Reply:
I don't think we would ever use this particular table for determining what the order of experiment enrollments was for a particular client. That being said, I think that what we want here would be closer to the latter (not their first ever experiment, but their experiment status when they were new)

Reply:
Sounds good, in that case the query is okay retrieving the first experiment and not updating it.